### PR TITLE
AWS ELB add support for ALPN policy details in NLB logs

### PR DESCRIPTION
**Pull request by agithomas**
- Bug 

## Proposed commit message

Add support for extracting ALPN policy related details from NLB access log data.  

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist
- [x] Update the new pattern in the pipeline test files and generate new expected log data
- [x] Package upgrade testing

## How to test this PR locally

```elastic-package build && elastic-package stack up -v -d --services package-registry```

## Related issues
- closes https://github.com/elastic/integrations/issues/11593

## Screenshots

image

### CODE REVIEW

File: packages/aws/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.30.2"
+  changes:
+    - description: Added the support for listeners with ALPN policy extension in ELB dataset for Network Load Balancers.
```
Comment (shmsr) [inline packages/aws/changelog.yml:4]:
```suggestion
    - description: Add the support for listeners with ALPN policy extension in ELB dataset for Network Load Balancers.
```


To match the tone with other entries in this file.

**Review by shmsr — APPROVED** [review]
Left one comment. Rest looks good!

File: packages/aws/changelog.yml (hunk 1 above)
Comment (agithomas) [reply packages/aws/changelog.yml]: Fixed. Marking as resolved.

**Review by gpop63 — APPROVED** [review]

**Review by lucian-ioan — APPROVED** [review]
LGTM

**Review by kcreddy — APPROVED** [review]
Adding code owner approval.

**Comment by agithomas** [comment]
@elastic/obs-ds-hosted-services , could you please review the PR?

File: packages/aws/data_stream/elb_logs/fields/fields.yml [hunk 2]
```diff
@@ -116,4 +116,23 @@
...
+    
+    - name: alpn_be_protocol
+      type: keyword
+      description: >
+        The application protocol negotiated with the target.
+    
+    - name: alpn_client_preference_list
+      type: keyword
+      description: >
+        The value of the application_layer_protocol_negotiation extension in the client hello message. This value is URL-encoded.
```
Comment (zmoog) [inline packages/aws/data_stream/elb_logs/fields/fields.yml:133]: Why did we keep the alpn_client_preference_list in its source form and not parse it as a list? Do we expect users to need it in its pristine form?

**Review by zmoog — APPROVED** [review]
LGTM, I left a question about `alpn_client_preference_list`.

File: packages/aws/data_stream/elb_logs/fields/fields.yml (hunk 2 above)
Comment (agithomas) [reply packages/aws/data_stream/elb_logs/fields/fields.yml:133]: In this is needed, we can do something like this. Keeping it for further reference / enhancement

```
  - script:
      lang: painless
      source: >-
        if (ctx?.aws?.elb?.alpn_client_preference_list != null && ctx?.aws?.elb?.alpn_client_preference_list != '-' && !ctx?.aws?.elb?.alpn_client_preference_list.trim().isEmpty()) {
          String cleanedList = ctx.aws.elb.alpn_client_preference_list;
          // Use regex to split by comma
          String[] protocols = /,/.split(cleanedList);

          ArrayList filteredProtocols = new ArrayList();
          for (String protocol : protocols) {
            def protocol_temp = /[\\"]/.matcher(protocol).replaceAll("");
            if (!protocol_temp.trim().isEmpty()) {
                filteredProtocols.add(protocol_temp.trim());
            }
          }
          ctx.aws.elb.alpn_client_preference_list = filteredProtocols;
        } else {
            // Set to an empty array if the conditions are not met
            ctx.aws.elb.alpn_client_preference_list = [];
        }
```

File: packages/aws/data_stream/elb_logs/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -221,6 +225,15 @@ processors:
       field:
         - _tmp
       ignore_missing: true
+  - date:
+      field: aws.elb.tls_connection_creation_time_str
+      target_field: aws.elb.tls_connection_creation_time
+      formats:  ["ISO8601"]
+      "if": "ctx?.aws?.elb?.tls_connection_creation_time_str != null && ctx?.aws?.elb?.tls_connection_creation_time_str != '-' && ctx?.aws?.elb?.tls_connection_creation_time_str != ''"
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/elb_logs/elasticsearch/ingest_pipeline/default.yml:232]:
```suggestion
      "if": "ctx.aws?.elb?.tls_connection_creation_time_str != null && ctx.aws?.elb?.tls_connection_creation_time_str != '-' && ctx.aws?.elb?.tls_connection_creation_time_str != ''"
```

**Pull request merged**